promdump/promdump.go

```diff
@@ -368,7 +381,113 @@ func exportMetric(ctx context.Context, promApi v1.API, metric string, beginTS ti
 			}
 		}
 	}
-	return writeFile(&values, filePrefix, fileNum)
+
+	return fileNum, writeFile(&values, filePrefix, fileNum)
+
+}
+
+func createArchive(buf io.Writer) error {
+	// Create new Writers for gzip , bzip2 and tar
+	// These writers are chained. Writing to the tar writer will
+
+	var tw *tar.Writer
+
+	switch *compression {
+	case "gzip":
+		gw := gzip.NewWriter(buf)
+		defer gw.Close()
+		tw = tar.NewWriter(gw)
+
+	case "bzip2":
+		// Add support for bzip2 compression
+		bw, err := bzip2.NewWriter(buf, nil) // Pass nil as the second argument, since bzip2 required to two argument so
+		if err != nil {
+			return err
+		}
+		defer bw.Close()
+		tw = tar.NewWriter(bw)
+	default:
+		return fmt.Errorf("unsupported compression type: %s", *compression)
+
+	}
+	defer tw.Close()
+
+	var filesToArchive []string
+
+	// Iterate over collectMetrics and add them to the tar archive
+	for _, v := range collectMetrics {
+		for i := 0; i < int(v.fileCount); i++ {
+			metricName, err := getMetricName(v)
+			if err != nil {
+				log.Fatalf("createArchive: %v", err)
+			}
+			filename := fmt.Sprintf("%s.%05d", metricName, i)
+			filesToArchive = append(filesToArchive, filename)
+		}
+	}
+	//custom filename collect nd add them to the tar archiv
+	if *metric != "" {
+		for i := 0; i < int(customMatricCount); i++ {
+			filename := fmt.Sprintf("%s.%05d", *metric, i)
+			filesToArchive = append(filesToArchive, filename)
+		}
+	}
+
+	for _, file := range filesToArchive {
+		// Add the file to the tar archive
+		err := addToArchive(tw, file)
+		if err != nil {
+			return err
+		}
+		// Remove the file after it has been added to the archive
+		err = os.Remove(file)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func addToArchive(tw *tar.Writer, filename string) error {
+	// Open the file which will be written into the archive
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Get FileInfo about our file providing file size, mode, etc.
+	info, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	// Create a tar Header from the FileInfo data
+	header, err := tar.FileInfoHeader(info, info.Name())
+	if err != nil {
+		return err
+	}
+
+	// Write file header to the tar archive
+	if err := tw.WriteHeader(header); err != nil {
+		return err
+	}
+
+	// Copy file content to tar archive
+	_, err = io.Copy(tw, file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func generateDefaultFilename() string {
+	compressionType := *compression
+	if compressionType == "gzip" {
+		return fmt.Sprintf("promdump-%s-%s.tar.gz", *nodePrefix, time.Now().Format("20060102-150405")) // Format: YYYYMMDD-HHMMSS
+	}
+	return fmt.Sprintf("promdump-%s-%s.bz2", *nodePrefix, time.Now().Format("20060102-150405"))
```

Comment:
Probably better as a `case` statement to match the code structure used for writer creation around L395.

Add handling for *compression = "none".

Style-wise, I think it would be better to have each `case` assign the string to a variable and have a single `return` for that variable at the end.